### PR TITLE
Test user login also in test suites with autologin

### PR DESCRIPTION
**Pull request by dzedro**
In openSUSE with autologin logout and login https://progress.opensuse.org/issues/13306

[gnome](http://10.100.12.155/tests/4597#step/first_boot/6)
[kde](http://10.100.12.155/tests/4598#step/first_boot/6)
[xfce](http://10.100.12.155/tests/4599#step/first_boot/7)
[lxde](http://10.100.12.155/tests/4600#step/first_boot/7)

### CODE REVIEW

File: tests/installation/first_boot.pm [hunk 1]
```diff
@@ -52,30 +76,24 @@ sub run() {
...
-    }
-    # the last check after previous intervals must be fatal
-    assert_screen \@tags, $check_interval;
-    if (match_has_tag('kde-greeter')) {
-        send_key "esc";
-        assert_screen 'generic-desktop';
+    else {
+        handle_first_desktop;
+        mouse_hide();    # hide mouse for clean logout needles
+                         # logout to test login on openSUSE https://progress.opensuse.org/issues/13306
```
Comment (okurz) [inline tests/installation/first_boot.pm]: that comment indendation is misleading, please move both comment lines above each statement, recheck and auto-LGTM :-)

Comment (dzedro) [inline tests/installation/first_boot.pm]: I had it so nice and the fast tidy at the end got me :P

**Comment by dzedro** [comment]
no I didn't but on SLE we don't have autologin, right ?

**Comment by dzedro** [comment]
I don't know about SLE test without [NOAUTOLOGIN](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L124)

**Comment by okurz** [comment]
right. There is https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/69ac0f36d00f865b9bf7fdf555a939a48fd1327c/tests/x11regressions/gnomecase/login_test.pm which uses different code AFAICS so I assume it's fine

**Comment by coolo** [comment]
Please needle 42.1: https://openqa.opensuse.org/tests/332836#step/first_boot/6 

CC @andreasstieger

File: tests/installation/first_boot.pm [hunk 2]
```diff
@@ -52,30 +76,25 @@ sub run() {
...
+            x11_start_program("$command");    # opens logout dialog
+            assert_screen 'logoutdialog' unless check_var('DESKTOP', 'gnome');
+        }
+        else {
+            my $key = check_var('DESKTOP', 'xfce') ? 'alt-f4' : 'ctrl-alt-delete';
+            send_key_until_needlematch 'logoutdialog', "$key";    # opens logout dialog
+        }
+        assert_and_click 'logout-button';                         # press logout
     }
+    handle_login;
```
Comment (DimStar77) [inline tests/installation/first_boot.pm]: That optimistically assumes the login prompt is already there right after the logout-button is clicked?

Especially in GNOME this fails as handle_login goes right to press 'RET" to select the user. But if gdm is not yet on screen/loaded, we fail to login afterwards..

See for example https://openqa.opensuse.org/tests/333191#step/first_boot/9 => it already expects the login prompt - but apparently the user selection was never done

Comment (dzedro) [inline tests/installation/first_boot.pm]: yes >< I have been very very optimistic

**Pull request merged**